skimage/measure/fit.py

```diff
@@ -274,27 +274,20 @@ def estimate(self, data):
         float_type = np.promote_types(data.dtype, np.float32)
         data = data.astype(float_type, copy=False)
 
-        x = data[:, 0]
-        y = data[:, 1]
+        # Adapted from a spherical estimator covered in a blog post by Charles Jeckel:
+        # https://jekel.me/2015/Least-Squares-Sphere-Fit/
+        A = np.append(data * 2, np.ones((data.shape[0], 1), dtype=float_type), axis=1)
+        f = np.sum(data ** 2, axis=1)
+        C, _, rank, _ = np.linalg.lstsq(A, f, rcond=None)
+
+        if rank != 3:
+            raise ValueError("Input data does not contain enough significant data points")
```

Comment:
I'm not sure about that. Isn't it more useful to raise an error rather than return useless estimates? If we change to a warning, are you suggesting we would also state in the message that this will become a ValueError in a future version?

Reply:
>  are you suggesting we would also state in the message that this will become a ValueError in a future version?

That seems in fact more reasonable :slightly_smiling_face:

Reply:
Okay, I have gone with this warning approach, adding a note to TODO.txt to remind us to switch to a ValueError in the future.